### PR TITLE
chore(tests): add tests for request unicode handling

### LINKED ISSUE #9: Ensure unicode handling is sane
Depends on #21. Write tests for unicode characters and make sure they pass.

### PR DESCRIPTION
**Pull request by brizental**
Fixes #9 

I just added some simple unicode tests for email address, subject and body.

I also added unicode character patterns in our email address regex, because the `.` alone didn't match for some reason...

r? @philbooth @vladikoff

### CODE REVIEW

File: src/validate/mod.rs [hunk 1]
```diff
@@ -23,21 +23,21 @@ lazy_static! {
     static ref AWS_ACCESS_FORMAT: Regex = Regex::new("^[A-Z0-9]+$").unwrap();
     static ref AWS_SECRET_FORMAT: Regex = Regex::new("^[A-Za-z0-9+/=]+$").unwrap();
     static ref BASE_URI_FORMAT: Regex = Regex::new(
-        "^https?://[A-Za-z0-9-]+(?:\\.[A-Za-z0-9-]+)*(?::[0-9]+)?/(?:[A-Za-z0-9-]+/)*$"
+        r"^https?://[A-Za-z0-9-]+(?:\.[A-Za-z0-9-]+)*(?::[0-9]+)?/(?:[A-Za-z0-9-]+/)*$"
```
Comment (philbooth) [inline src/validate/mod.rs]: Oh cool, I didn't know regex literals are a thing!

File: src/send/test.rs [hunk 2]
```diff
@@ -122,6 +123,82 @@ fn without_optional_data() {
...
+      "provider": "mock"
+    }"#,
+        )
+        .dispatch();
+
+    assert_eq!(response.status(), Status::Ok);
+
+    let body = response.body().unwrap().into_string().unwrap();
+    assert_eq!(body, json!({ "messageId": "mock:deadbeef" }).to_string());
+}
```
Comment (philbooth) [inline src/send/test.rs]: These tests are great but is it also possible to add some unicode tests to `src/providers/ses/test.rs`, to make sure that there's no surprises for us at that level?

File: src/validate/mod.rs (hunk 1 above)
Comment (brizental) [reply src/validate/mod.rs]: Yeah, it wouldn't let me do de `\pL` otherwise, said that `\p` wasn't a character they could escape.

Comment (brizental) [reply src/validate/mod.rs]: I changed all the other ones as well to avoid that problem and also all the examples in the `Regex` docs are like that (https://docs.rs/regex/1.0.2/regex/)
